### ISSUE TITLE
Arrow function parameters containing comment has parens removed incorrectly

### ISSUE THREAD

**Issue by jakebailey** [dprint/dprint-plugin-typescript#533]
**Describe the bug**

dprint-plugin-typescript version: 0.85.0

**Input Code**

```ts
client
    .send(lookup)
    .then((
        resp, // $ExpectType Batch<Lookup>
    ) => console.log(resp))
    .catch((
        err, // $ExpectType any
    ) => console.error(err));
```

Config:

```json
{
  "arrowFunction.useParentheses": "preferNone"
}
```

**Expected Output**

```ts
client
    .send(lookup)
    .then((
        resp, // $ExpectType Batch<Lookup>
    ) => console.log(resp))
    .catch((
        err, // $ExpectType any
    ) => console.error(err));
```

(or something)

**Actual Output**

```ts
client
    .send(lookup)
    .then(resp // $ExpectType Batch<Lookup>
     => console.log(resp))
    .catch(err // $ExpectType any
     => console.error(err));
```

Hit roughly 4 times in testing DT with dprint.

**Comment by dsherret** [comment]
The invalid output can be reduced to:

```ts
call(arg
=> 5);
```

It seems typescript is able to parse this, but swc, babel and chrome can't, so I'm going to count this as a dprint bug. It should be keeping the parens in this case.

**Comment by dsherret** [comment]
I actually had a condition sort of for this already, but it didn't handle a trailing commas on the parameter 😅.

Fixed in 0.85.1.
